docsrc/antora-extension/index.js

```diff
@@ -0,0 +1,172 @@
+      rawData.annotations.forEach((a) => {
+
+        const isPackageScope = a.annotations.scope == "package"
+        const isRuleScope = a.annotations.scope == "rule"
+
+        const dottedPath = a.path.map((i) => i.value).join(".")
+        const isDeny = dottedPath.endsWith(".deny")
+        const isWarn = dottedPath.endsWith(".warn")
+        const isPolicy = dottedPath.startsWith("data.policy.")
+
+        // If we see a package scoped annotation then save it
+        if (isPackageScope) {
+          pkgAnnotation = a.annotations
+        }
+
+        // Anything that isn't a top level rule doesn't get documented
+        if (isPolicy && (isDeny || isWarn) && isRuleScope) {
+          const title = a.annotations.title
+          const description = a.annotations.description
+          const shortName = a.annotations.custom.short_name
+          const warningOrFailure = isDeny ? "warning" : "failure"
+          const failureMsg = a.annotations.custom.failure_msg
+          const effectiveOn = a.annotations.custom.effective_on
+          const file = a.location.file
+          const row = a.location.row
+
+          // The package name, will be used for grouping
+          const pkgHeading = a.path[3].value
+
+          // If there is some package scoped rule data then merge it in to the rule scoped rule data
+          var ruleData = a.annotations.custom.rule_data
+          if (pkgAnnotation.custom && pkgAnnotation.custom[shortName] && pkgAnnotation.custom[shortName].rule_data) {
+            ruleData = {...ruleData, ...pkgAnnotation.custom[shortName].rule_data}
+          }
+
+          data.push({
+            dottedPath,
+            pkgHeading,
+            shortName,
+            title,
+            description,
+            ruleData,
+            warningOrFailure,
+            failureMsg,
+            effectiveOn,
+            file,
+            row,
+          })
+        }
+      })
+
+      // Find the template for generating the rules documentation
+      const ruleDocsHbs = content.files.find(f => f.path.endsWith('rules.adoc.hbs'))
+      const template = Handlebars.compile(ruleDocsHbs.contents.toString())
+
+      // To make life easier let's hard code the two policies like this
+      const policies = [
+        {
+          title: "Release Policy",
+          stem: "release",
+          namespace: "data.policy.release",
+          intro: "These rules are applied to pipeline run attestations associated with container images built by HACBS.",
+        },
+
+        {
+          title: "Pipeline Policy",
+          stem: "pipeline",
+          namespace: "data.policy.pipeline",
+          intro: "These rules are applied to Tekton pipeline definitions.",
+        },
+      ]
+
+      // Similar for the acceptable bundles page
+      // TODO
+      const acceptable_bundles = [
+        {
+          title: "Acceptable Bundles",
+          stem: "acceptable_bundlesrelease",
+          intro: "...",
+        },
+      ]
```

Comment:
Perhaps we should have this in Rego annotations (for future PR)

Reply:
This went away in later revisions. The content is now in the hbs template which I think is a better place for it.

Reply:
(That said, we could consider using annotations for the text at some point.)